Make the program naming scheme uniform w.r.t. the standard one. (#15442)
coq/coq: doc/changelog/07-vernac-commands-and-options/15442-deprecate-program-naming-mode.rst

- **Changed:**

>   The algorithm for name generation of anonymous variables
>   for ``Program`` subproofs was made uniform with the one
>   used in the general case. This can create incompatibilities

Reviewer: ```suggestion
  The algorithm for name generation of anonymous variables
  for ``Program`` subproofs is now the same as the one that is
  generally used. This can create incompatibilities
```
